pkg/asset/agent/configimage/configiso.go

```diff
@@ -0,0 +1,159 @@
+package configimage
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/vincent-petithory/dataurl"
+
+	"github.com/openshift/assisted-image-service/pkg/isoeditor"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/agent/image"
+)
+
+const (
+	configImageFilename = "agentconfig.noarch.iso"
+	configCpioArchive   = "config.gz"
+
+	pullSecretManifest          string = "/etc/assisted/manifests/pull-secret.yaml" //nolint:gosec // not hardcoded credentials
+	nmStateConfigManifest       string = "/etc/assisted/manifests/nmstateconfig.yaml"
+	clusterDeploymentManifest   string = "/etc/assisted/manifests/cluster-deployment.yaml"
+	clusterImageSetManifest     string = "/etc/assisted/manifests/cluster-image-set.yaml"
+	agentClusterInstallManifest string = "/etc/assisted/manifests/agent-cluster-install.yaml"
+	extraManifests              string = "/etc/assisted/extra-manifests"
+	hostConfigFiles             string = "/etc/assisted/hostconfig"
+	hostNameFiles               string = "/etc/assisted/hostnames"
+	kubeadminPasswordHash       string = "/opt/agent/tls/kubeadmin-password.hash" //nolint:gosec // not hardcoded credentials
+	kubeconfigTLSCerts          string = "/opt/agent/tls/admin-kubeconfig-signer"
+	kubeapiserverTLSCerts       string = "/opt/agent/tls/kube-apiserver"
+)
+
+// TODO - need to add rendezvous host file when available.
+var fileFilters = []string{
+	pullSecretManifest,
+	nmStateConfigManifest,
+	clusterDeploymentManifest,
+	clusterImageSetManifest,
+	agentClusterInstallManifest,
+	extraManifests,
+	hostConfigFiles,
+	hostNameFiles,
+	kubeadminPasswordHash,
+	kubeconfigTLSCerts,
+	kubeapiserverTLSCerts,
+}
```

Comment:
When is the kubeconfig file being created in this flow? How is it exposed to the user? Using console/ssh/etc?

Reply:
In `auth/` (after last update)